src/main/java/com/joshdev/deathmanager/DeathManager.kt

```diff
@@ -0,0 +1,61 @@
+/* Licensed under GNU General Public License v3.0 */
+package com.joshdev.deathmanager
+
+import com.joshdev.deathmanager.commands.RestoreInventory
+import com.joshdev.deathmanager.commands.ShowDeaths
+import com.joshdev.deathmanager.events.OnPlayerDeath
+import com.joshdev.deathmanager.exceptions.DeathManagerException
+import org.bukkit.plugin.java.JavaPlugin
+import java.io.File
+import java.sql.Connection
+import java.sql.DriverManager
+import java.util.logging.Logger
+
+@Suppress("UNUSED") // Main class comes up as unused when it actually is, Kotlin dumb si
+class DeathManager : JavaPlugin() {
+
+    companion object {
+        lateinit var pluginLogger: Logger
+            private set
+        lateinit var dbConnection: Connection
+            private set
+        lateinit var pluginInstance: DeathManager
+            private set
+        fun isLoggerInitialized(): Boolean {
+            return ::pluginLogger.isInitialized
+        }
+        fun isConnectionInitialized(): Boolean {
+            return ::dbConnection.isInitialized
+        }
+        fun isPluginInstanceInitialized(): Boolean {
+            return ::pluginInstance.isInitialized
+        }
+    }
+
+    override fun onEnable() {
+        logger.info("Setting up DeathManager...")
+        pluginLogger = logger
+        pluginInstance = this
+        if (!dataFolder.exists()) {
+            dataFolder.mkdir()
+        }
+        val dbPath = dataFolder.absolutePath + "/data.db"
+        val dbFile = File(dbPath)
+        if (!dbFile.exists()) {
+            dbFile.createNewFile()
+        }
+        val connection = DriverManager.getConnection("jdbc:sqlite:$dbPath") ?: throw DeathManagerException("Connection to database couldn't be established.")
+        dbConnection = connection
+        val stmt = connection.createStatement()
+        stmt.execute("CREATE TABLE IF NOT EXISTS deaths(id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT, uniqueId TEXT NOT NULL, timeOfDeath INTEGER NOT NULL, posX REAL NOT NULL, posY REAL NOT NULL, posZ REAL NOT NULL, worldName TEXT NOT NULL, serializedInventory BLOB NOT NULL);")
```

Comment:
1. You should use try with resources on SQL statements, and connection.
2. The SQL statements should be in a single object (often might make object per table).
You probably want to create DB wrapper which contains the connection and store it in place of current connection.
One of the things it does is it would remove duplication like:
https://github.com/NearVanilla/DeathDB/pull/2/files#diff-ede5b7b85a3fa1be3530ddbc9eebd660833cada162e9d31280249841ed5de6efR74
https://github.com/NearVanilla/DeathDB/pull/2/files#diff-a30b8cb9dd90cd2993eb3319faf43739f73d443337f96035177ebaed609dceffR42

Reply:
I did actually look back at my code some stage after bringing these changes to your attention and I realized I should have done your first point. Not to worry, that will be resolved.

I'll also look into creating a DB Wrapper as described. I'll update as development progresses.

Reply:
Resolved in some capacity.